stdlib/src/collections/list.mojo

```diff
@@ -604,3 +602,20 @@ struct List[T: CollectionElement](CollectionElement, Sized, Boolable):
                 result += ", "
         result += "]"
         return result
+
+    @staticmethod
+    fn count[T: ComparableCollectionElement](self: List[T], value: T) -> Int:
+        """Counts the number of occurrences of a value in the list.
```

Comment:
You should add a small example explaining how to use this method in the docstring. It might not be intuitive for users to use `__type_of(...)`.

Reply:
@gabrieldemarmiesse should the note be similar to the __str__ method?

```Note that since we can't condition methods on a trait yet,
the way to call this method is a bit special. Here is an example below:

`mojo
var my_list = List[Int](1, 2, 3)
print(__type_of(my_list).count(my_list, 1))
`

When the compiler supports conditional methods, then a simple `my_list.count(1)` will
be enough.```

Reply:
yes that looks good :)

Reply:
Pushed the changes